mathesar/utils/tables.py

```diff
@@ -16,17 +22,37 @@ def get_table_column_types(table):
     return col_types
 
 
-def create_empty_table(data):
+def create_table_from_data(request, data):
```

Comment:
I think this should go directly in the viewset instead of being a separate function/file, since you're using DRF responses.